sql/repl_failsafe.cc

```diff
@@ -191,6 +184,10 @@ static my_bool show_slave_hosts_callback(THD *thd, Protocol *protocol)
     }
     protocol->store((uint32) si->port);
     protocol->store(si->master_id);
+    protocol->store(si->log_file, safe_strlen(si->log_file), &my_charset_bin);
+    protocol->store(si->log_pos);
+    protocol->store(gtid);
+    repl_semisync_master.store_status(protocol);
```

Comment:
I think it would make more sense here to use `rpl_semi_sync_master_status` to avoid double-locking as well as the additional dependency additions.